cmd/common/utils/parser-tableformatter.go

```diff
@@ -97,3 +103,27 @@ func (p *GadgetParser[T]) BuildColumnsHeader() string {
 func (p *GadgetParser[T]) TransformToColumns(entry *T) string {
 	return p.formatter.FormatEntry(entry)
 }
+
+func (p *GadgetParser[T]) TransformToTable(entries []*T) string {
+	maxWidth := 0
+	if term.IsTerminal(int(os.Stdout.Fd())) {
+		terminalWidth, _, err := term.GetSize(0)
+		if err == nil {
+			maxWidth = terminalWidth
+		}
+	}
+
+	// Disable auto-scaling as AdjustWidthsToContent will already manage the
+	// screen size with the maxWidth we are passing.
+	// p.formatter.SetAutoScaling(false)
+
+	p.formatter.AdjustWidthsToContent(entries, true, maxWidth, true)
+
+	var sb strings.Builder
+	p.formatter.WriteTable(&sb, entries)
+	return sb.String()
```

Comment:
I think `WriteTable` should follow the same logic as `FormatEntry` and `FormatHeader`. I meant, it should be `FormatTable` and should not print the table.

Notice also that `FormatTable` should not add a new line at the end of the table, like `FormatEntry` and `FormatHeader`.

Reply:
Solve using https://github.com/kinvolk/inspektor-gadget/pull/993.